timeboost-contract/src/committee.rs

```diff
@@ -0,0 +1,234 @@
+//! Committee management and querying utilities
+use crate::bindings::keymanager::KeyManager;
+use alloy::{primitives::Address, providers::Provider};
+use anyhow::Result;
+
+pub struct CommitteeManager<P> {
+    provider: P,
+    contract_addr: Address,
+}
+
+impl<P: Provider> CommitteeManager<P> {
+    pub fn new(provider: P, contract_addr: Address) -> Self {
+        Self {
+            provider,
+            contract_addr,
+        }
+    }
+
+    pub async fn get_committees_for_startup(
+        &self,
+        current_id: u64,
+        previous_id: Option<u64>,
+    ) -> Result<(KeyManager::Committee, Option<KeyManager::Committee>)> {
```

Comment:
i find `struct CommitteeManager` an overkill, and i also find this function a bit unnecessary. 

usually wherever i need to fetch committee info, i simply have two lines, one line to get a `KeyManager` contract instance, another line to `.getCommitteeById()` on it. that's it. 

if we use this struct, we first need to `CommitteManager::new()`, then call this `for_startup()`. 
imo, the former is clearer and less indirection.


The decision to fetch "which committee" during startup could be logic that resides in `timeboost` code, not here. 
I vote for dropping this file.